components/micro-gateway-cli/src/main/resources/templates/endpointReinit.mustache

```diff
@@ -22,28 +23,30 @@
             }
         }
                         if (reinitRequired) {
-                        //destination_attribute = <string>gateway:etcdUrls[etcdKey];
+                        map<string> etcdUrls = gateway:getEtcdUrlsMap();
+                        //destination_attribute = <string>etcdUrls[etcdKey];
                         {{#equals type "http"}} {{>httpEndpointReInit}} {{/equals}}
                         {{#equals type "load_balance"}} {{>lbEndpointReInit}} {{/equals}}
                         {{#equals type "failover"}} {{>failoverEndpointReInit}} {{/equals}}
 
-                            if(err is error) {
+                            if(reInitializedEndpoint is error) {
                                 reinitFailed = true;
-                                gateway:urlChanged[failedEtcdKey] = true;
+                                map<any> urlChangedMap = gateway:getUrlChangedMap();
+                                urlChangedMap[failedEtcdKey] = true;
 
                                 http:Response res = new;
                                 res.statusCode = 500;
                                 json payload = {
                                     "fault": {
                                         "code": "101503",
                                         "message": "Runtime Error",
-                                        "description": "Error connecting to the back end"
+                                        "description": "Error while connecting to re initialized backend"
                                     }
                                 };
                                 invocationContext.attributes["error_code"] = "101503";
                                 res.setPayload(payload);
                                 clientResponse = res;
-                                log:printError("URL defined at etcd for key " + config:getAsString(failedEtcdKeyConfigValue) + " is invalid");
+                                log:printError("URL defined at etcd for key " + gateway:getConfigValue("", failedEtcdKeyConfigValue, "") + " is invalid");
```

Comment:
Two spaces needed to print like below.
URL defined at etcd for key <key> is invalid. 
Other wise it will print like
URL defined at etcd for key <key>is invalid.

Reply:
Didn't see that. My bad :)